spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/config/MongoDbOutboundGatewayParserTests.java

```diff
@@ -109,6 +110,20 @@ public void fullConfigWithMongoTemplate() {
 		assertEquals("foo", TestUtils.getPropertyValue(gateway, "collectionNameExpression.literalValue"));
 	}
 
+	@Test
+	public void fullConfigWithCollectionCallback() {
+		MongoDbOutboundGateway gateway = TestUtils.getPropertyValue(
+				context.getBean("fullConfigWithCollectionCallback"), "handler", MongoDbOutboundGateway.class);
+
+		assertEquals(context.getBean("mongoDbTemplate"), TestUtils.getPropertyValue(gateway, "mongoTemplate"));
+		assertNull(TestUtils.getPropertyValue(gateway, "mongoDbFactory"));
+		assertNull(TestUtils.getPropertyValue(gateway, "mongoConverter"));
+		assertNotNull(TestUtils.getPropertyValue(gateway, "evaluationContext"));
+		assertTrue(TestUtils.getPropertyValue(gateway, "collectionNameExpression") instanceof LiteralExpression);
```

Comment:
There is `assertThat(..., instanceOf(...))`. 
Much better, if an actual isn't of expected type.